src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,82 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Represents the FindCommand in the Duke program.
+ */
+public class FindCommand extends Command {
+    private final String keyword;
+
+    /**
+     * Construct a FindCommand to find all task(s) with the given keyword in the list of tasks.
+     *
+     * @param keyword Keyword to be used to find all task(s) in the list of tasks.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Defines the execution of the FindCommand where all task(s) with the given keyword is/are listed.
+     *
+     * @param tasks Tasks of the Duke program.
+     * @param ui Ui of the Duke program.
+     * @param storage Storage of the Duke program.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        TaskList matchingTasks = new TaskList();
+        for (int i = 0; i < tasks.getSize(); i++) {
+            Task currTask = tasks.getTask(i);
```

Comment:
You might prefer to use the enhanced `for` loop here, i.e. `for (Task currTask : tasks)`. (This also applies to the loops in other command classes.)

Reply:
Yes, thank you for the suggestion!